test/util.go

```diff
@@ -86,6 +122,12 @@ func TapSecondaryAt(obj fyne.SecondaryTappable, pos fyne.Position) {
 	obj.TappedSecondary(ev)
 }
 
+// TryToTapCanvas tries to tap at an absolute position on the canvas.
```

Comment:
When I introduced `TapCanvas` I first wanted to ignore the not-found. But then I realised that in most cases I expected a certain element to be there and that a test failure was a good help to show if I got a hit. And I didn’t want to return `bool` and write
```go
  if !test.TapCanvas(…) {
    t.Error(…)
  }
```
every time.

Reply:
Returning a bool and having an error both seem like side effects of the main objective.
The code would be more explicit if there was an "ObjectAtPosition" or "ObjectAtPositionMatching" like we have in internal code.

But the more I think about it the real test should be "i tapped this position, did I get the outcome I was expecting" which does not require any of the above.

Reply:
Yes, but there is at least one test which assumes that nothing happens when tapping at a certain (tappable) position: TestMenu_TappedPaddingOrSeparator

Reply:
If not-tappable would be silently ignored this test could be false-positive.

Reply:
I read through that test but I don't understand what you mean. It is checking for the resulting behaviour of taps - not erroring when the item is not tappable seems entirely reasonable.

If a test utility was expected to fail with then I think we should have included it in the name - i.e. AssertTappableAndTap
I had expected that the test utilities help drive tests rather than to make assertions for the user's code.

If this assert code is really needed then can I propose dropping the "TryTo" prefix for non-erroring methods (otherwise "Type" should be "TryToType" etc) and instead add a more explicit failing indicator to those tests that we want to issue assertions etc...